pkg/controllers/subnetbinding/controller.go

```diff
@@ -0,0 +1,353 @@
+		return nil, &errorWithRetry{
+			message: fmt.Sprintf("Subnet CR %s is not realized on NSX", name),
+			retry:   false,
+			error:   fmt.Errorf("not found NSX VpcSubnets created by Subnet CR '%s/%s'", namespace, name),
+		}
+	}
+
+	// Check if the Subnet CR is nested.
+	if !isTarget {
+		bms := r.SubnetBindingService.GetSubnetConnectionBindingMapsByParentSubnet(subnets[0])
+		if len(bms) > 0 {
+			dependency := r.SubnetBindingService.GetCRNameBySubnetConnectionBindingMap(bms[0])
+			return nil, &errorWithRetry{
+				message: fmt.Sprintf("Subnet CR %s is working as target by %s", name, dependency),
+				error:   fmt.Errorf("Subnet %s already works as target in SegmentConnectionBindingMap %s", name, dependency),
+				retry:   true,
+			}
+		}
+	} else {
+		bms := r.SubnetBindingService.GetSubnetConnectionBindingMapsByChildSubnet(subnets[0])
+		if len(bms) > 0 {
+			dependency := r.SubnetBindingService.GetCRNameBySubnetConnectionBindingMap(bms[0])
+			return nil, &errorWithRetry{
+				message: fmt.Sprintf("Target Subnet CR %s is attached by %s", name, dependency),
+				error:   fmt.Errorf("target Subnet %s is already attached by SegmentConnectionBindingMap %s", name, dependency),
+				retry:   true,
+			}
+		}
+	}
+
+	return subnets, nil
+}
+
+func (r *Reconciler) validateVpcSubnetsBySubnetSetCR(ctx context.Context, namespace, name string) ([]*model.VpcSubnet, *errorWithRetry) {
+	subnetSetCR := &v1alpha1.SubnetSet{}
+	subnetSetKey := types.NamespacedName{Namespace: namespace, Name: name}
+	err := r.Client.Get(ctx, subnetSetKey, subnetSetCR)
+	if err != nil {
+		log.Error(err, "Failed to get SubnetSet CR", "key", subnetSetKey.String())
+		return nil, &errorWithRetry{
+			message: fmt.Sprintf("Unable to get SubnetSet CR %s", name),
+			error:   fmt.Errorf("failed to get SubnetSet %s in Namespace %s with error: %v", name, namespace, err),
+			retry:   false,
+		}
+	}
+
+	subnets := r.SubnetService.ListSubnetCreatedBySubnetSet(string(subnetSetCR.UID))
+	if len(subnets) == 0 {
+		log.Info("NSX VpcSubnets by SubnetSet CR do not exist", "key", subnetSetKey.String())
+		return nil, &errorWithRetry{
+			message: fmt.Sprintf("SubnetSet CR %s is not realized on NSX", name),
+			error:   fmt.Errorf("no existing NSX VpcSubnet created by SubnetSet CR '%s/%s'", namespace, name),
+			retry:   false,
+		}
+	}
+	return subnets, nil
+}
+
+func updateBindingMapStatusWithUnreadyCondition(c client.Client, ctx context.Context, obj client.Object, _ metav1.Time, _ error, args ...interface{}) {
+	bindingMap := obj.(*v1alpha1.SubnetConnectionBindingMap)
+	reason := args[0].(string)
+	msg := args[1].(string)
+	condition := v1alpha1.Condition{
+		Type:    v1alpha1.Ready,
+		Status:  corev1.ConditionFalse,
+		Reason:  reason,
+		Message: msg,
+	}
+	updateBindingMapCondition(c, ctx, bindingMap, condition)
+}
+
+func updateBindingMapStatusWithReadyCondition(c client.Client, ctx context.Context, obj client.Object, _ metav1.Time, _ ...interface{}) {
```

Comment:
Why not to define `func(client.Client, context.Context, client.Object, metav1.Time, ...interface{}){}` in the caller
and leave used parameters only in this function.